### PR TITLE
ES DQM: Enable concurrent processing of lumisections

### PR DESCRIPTION
**Pull request by tanmaymudholkar**
This fix enables concurrent processing of lumisections with ESIntegrityTask included in the workflow.

#### PR description:

This PR changes ESIntegrityTask from a `DQMOneLumiEDAnalyzer` to a `DQMOneEDAnalyzer` to enable concurrent processing of lumisections.

#### PR validation:
The standard suite of tests was run: `runTheMatrix.py -l limited -i all --ibeos`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport.

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test with #31699
To enable concurrent LS processing

File: DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc [hunk 1]
```diff
@@ -67,9 +67,12 @@ void ESIntegrityTask::dqmEndRun(const Run& r, const EventSetup& c) {
   // TODO: no longer possible, clone histo beforehand if full statisticcs at end of run are required.
 }
 
-void ESIntegrityTask::dqmBeginLuminosityBlock(const edm::LuminosityBlock& lumi, const edm::EventSetup& c) {
+std::shared_ptr<ESLSCache> ESIntegrityTask::globalBeginLuminosityBlock(const edm::LuminosityBlock& lumi,
+                                                                       const edm::EventSetup& c) const {
   LogInfo("ESIntegrityTask") << "analyzed " << ievt_ << " events";
   // In case of Lumi based analysis SoftReset the Integrity histogram
+  auto lumiCache = std::make_shared<ESLSCache>();
+  lumiCache->ievtLS_ = 0;
```
Comment (Dr15Jones) [inline DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc:75]: Below here you reset the MonitorElements. That behavior is incompatible with concurrent Lumis as you are sharing the same MonitorElement so you reset it while events of the previous Lumi are still being processed.

Comment (tanmaymudholkar) [reply DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc:75]: @Dr15Jones In that case, is it impossible to have per-lumisection resets with concurrent processing? What if the MonitorElement were part of the luminosity cache (book a new histogram for each lumisection) -- would that work? Or is there another officially supported solution? Sorry, one reason this fix was difficult for ECAL was I tried to look through other subsystems' code and couldn't find these resets...

**Comment by Dr15Jones** [comment]
@cms-sw/dqm-l2 your input is needed to understand how to properly handle per lumi MonitorElements.

**Comment by tanmaymudholkar** [comment]
@Dr15Jones @jfernan2 I've pushed a fix for this PR that removes the need to reset the ME in question. The four plots "ESIntegrityTask/ES Integrity Errors Z X P Y" are now no longer reset per lumisection, so we may see a difference in offline tests, but this is OK because of the new way we calculate the number of errors per lumisection.

**Comment by jfernan2** [comment]
please test with #31699

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**